plumbing/format/diff/patch.go

```diff
@@ -0,0 +1,72 @@
+package diff
+
+import (
+	"io"
+
+	"gopkg.in/src-d/go-git.v4/plumbing"
+	"gopkg.in/src-d/go-git.v4/plumbing/filemode"
+)
+
+// Operation defines the operation of a diff item.
+type Operation int
+
+const (
+	// Equal item represents a equals diff.
+	Equal Operation = iota
+	// Add item represents an insert diff.
+	Add
+	// Delete item represents a delete diff.
+	Delete
+)
+
+// Patch represents the necessary steps to transform several files to another.
+type Patch interface {
+	// FilePatches returns a slice of patches per file.
+	FilePatches() []FilePatch
+	// Message returns an optional message that can be at the top of the
+	// Patch representation.
+	Message() string
+}
+
+// Patcheable contains ways to represent a patch. Its use is intended to create
+// fluents APIs, and be able to discover more easily ways to work with Patches.
+// You don't need to know that a NewUnifiedEncoder constructor exists, you just
+// need to see the available methods after get a Patch.
```

Comment:
The idea of this interface is do this:
```go
p, err := commit.Patch(otherCommit)
handleErr(err)
err = p.UnifiedDiff(buf,2)
handleErr(err)
```
Instead of: 
```go
p, err := commit.Patch(otherCommit)
handleErr(err)
// To do this, you need to know the internals or find it into the documentation
ue := diff.NewUnifiedEncoder(buf, 1)
err = ue.Encode(p)
handleErr(err)
```